fivetran_provider_async/hooks.py

```diff
@@ -134,6 +147,47 @@ async def get_sync_status_async(self, connector_id, previous_completed_at):
             job_status = "pending"
             return job_status
 
+    def pause_and_restart(self, connector_id, reschedule_for, reschedule_time):
+        """
+        While a connector is syncing, if it falls into a reschedule state,
+        wait for a time either specified by the user of recommended by Fivetran,
+        Then restart a sync
+
+        :param connector_id: Fivetran connector_id, found in connector settings
+            page in the Fivetran user interface.
+        :type connector_id: str
+        :param reschedule_for: From connector details, if schedule_type is manual,
+            then the connector expects triggering the event at the designated UTC time
+        :type reschedule_for: str
+        :param reschedule_time: Optional, if connector is in reset state
+            number of seconds to wait before restarting, else Fivetran suggestion used
+        :type reschedule_time: int
+        """
+        if reschedule_time:
+            log_statement = f'Starting connector again in "{reschedule_time}" seconds'
+            self.log.info(log_statement)
+            time.sleep(reschedule_time)
+        else:
+            wait_time = (
+                self._parse_timestamp(reschedule_for).add(minutes=1) - pendulum.now(tz="UTC")
```

Comment:
Why do we add minutes=1 here?

Reply:
We add 1 minute to the time suggested from Fivetran so that we have buffer when we make API call